test/tools/ossfuzz/SolidityEvmoneInterface.cpp

```diff
@@ -0,0 +1,193 @@
+	std::string const& _contractName,
+	std::map<std::string, solidity::util::h160> _libraryAddresses,
+	frontend::OptimiserSettings _optimization
+)
+{
+	std::string sourceCode = _sourceCode;
+	m_compiler.setSources({{"", sourceCode}});
+	m_compiler.setLibraries(move(_libraryAddresses));
+	m_compiler.setEVMVersion(m_evmVersion);
+	m_compiler.setOptimiserSettings(_optimization);
+	if (!m_compiler.compile())
+	{
+		langutil::SourceReferenceFormatter formatter(std::cerr, false, false);
+
+		for (auto const& error: m_compiler.errors())
+			formatter.printExceptionInformation(
+					*error,
+					formatter.formatErrorInformation(*error)
+			);
+		std::cerr << "Compiling contract failed" << std::endl;
+	}
+	evmasm::LinkerObject obj = m_compiler.object(
+		_contractName.empty() ?
+		m_compiler.lastContractName() :
+		_contractName
+	);
+	return obj.bytecode;
+}
+
+bool EvmoneUtility::isOutputExpected(
+	uint8_t const* _result,
+	size_t _length,
+	std::vector<uint8_t> const& _expectedOutput
+)
+{
+	if (_length != _expectedOutput.size())
+		return false;
+
+	return (memcmp(_result, _expectedOutput.data(), _length) == 0);
+}
+
+evmc_message EvmoneUtility::initializeMessage(bytes const& _input)
+{
+	// Zero initialize all message fields
+	evmc_message msg = {};
+	// Gas available (value of type int64_t) is set to its maximum
+	// value.
+	msg.gas = std::numeric_limits<int64_t>::max();
+	msg.input_data = _input.data();
+	msg.input_size = _input.size();
+	return msg;
+}
+
+evmc::result EvmoneUtility::executeContract(
+	EVMHost& _hostContext,
+	bytes const& _functionHash,
+	evmc_address _deployedAddress
+)
+{
+	evmc_message message = initializeMessage(_functionHash);
+	message.destination = _deployedAddress;
+	message.kind = EVMC_CALL;
+	return _hostContext.call(message);
+}
+
+evmc::result EvmoneUtility::deployContract(EVMHost& _hostContext, bytes const& _code)
+{
+	evmc_message message = initializeMessage(_code);
+	message.kind = EVMC_CREATE;
+	return _hostContext.call(message);
+}
+
+evmc::result EvmoneUtility::deployAndExecute(EVMHost& _hostContext, bytes _byteCode, std::string _hexEncodedInput)
+{
+	// Deploy contract and signal failure if deploy failed
+	evmc::result createResult = deployContract(_hostContext, _byteCode);
+	solAssert(
+		createResult.status_code == EVMC_SUCCESS,
+		"Proto solc fuzzer: Contract creation failed"
+	);
+
+	// Execute test function and signal failure if EVM reverted or
+	// did not return expected output on successful execution.
+	evmc::result callResult = executeContract(
+		_hostContext,
+		util::fromHex(_hexEncodedInput),
+		createResult.create_address
+	);
+
+	// We don't care about EVM One failures other than EVMC_REVERT
+	solAssert(callResult.status_code != EVMC_REVERT, "Proto solc fuzzer: EVM One reverted");
+	return callResult;
+}
+
+evmc::result EvmoneUtility::compileDeployAndExecute(
+	EVMHost& _hostContext,
+	std::string _sourceCode,
+	std::string _contractName,
+	std::string _methodName,
```

Comment:
Would it make sense for these to be `const&`? I guess `_sourceCode` at least can get large?